admin_guide/etcd_migration.adoc

```diff
@@ -0,0 +1,138 @@
+the migration process must wait until all members converge.
+
+Additionally, the migration should not be run repeatedly, as new v2 data can overwrite already migrated v3 data.
+
+== Migration
+
+The migration itself is performed by running the following command (with etcd daemon stopped) for each etcd member:
+
+----
+ETCDCTL_API=3 etcdctl migrate --data-dir=/var/lib/etcd
+----
+
+The `--data-dir` can point to different location depending on the deployment.
+For example, embedded etcd operates over `/var/lib/origin/openshift.local.etcd` directory,
+etcd run as a system container operates over `/var/lib/etcd/etcd.etcd` directory.
+
+Once done the migration responds with success:
+
+----
+finished transforming keys
+----
+
+In case there are no v2 data with:
+
+----
+no v2 keys to migrate
+----
+
+Or with a failure.
+
+If `--no-ttl` option of the `etcdctl migrate` command is not specified, TLL keys are migrated as well.
+Given the TTL keys in v2 data are replaced with leases in v3 data,
+one needs to attach leases to all migrated TTL keys by running (with etcd daemon running):
+
+----
+oadm migrate etcd-ttl <certificate details> <endpoint> --ttl-keys-prefix '/kubernetes.io/events' --lease-duration 1h
+----
+
+Additionally, for HA deployments the command needs to be run one more time with `--ttl-keys-prefix` set to `/kubernetes.io/masterleases`.
+
+The command (available since `atomic-openshift-3.6.111-1`) needs to be run only once on any of currently available masters.
+
+Once the migration and the lease attachment is done, it is recommended to validate the v3 data.
+Given the v3 key space has no concept of directories,
+it is enough to check that all v2 non-directory keys are available in v3 database and
+all their values are equal.
+
+== Master re-configuration
+
+Once the migration is done,
+the xref:../install_config/master_node_configuration.adoc#master-configuration-files[master configuration file] (the *_/etc/origin/master/master-config.yaml_* file by default)
+needs to be updated so the master daemons can use the new storage backend:
+
+====
+[source,yaml]
+----
+kubernetesMasterConfig:
+  apiServerArguments:
+    storage-backend:
+    - etcd3
+    storage-media-type:
+    - application/vnd.kubernetes.protobuf
+----
+====
+
+== Migration steps
+
+1. run pre-migration checks (etcd cluster is healthy, raft indices are converged, no v3 data available)
+1. backup etcd v2 data
+1. stop master daemons and etcd members
+1. migrate v2 data to v3
+1. start etcd members
+1. attach leases (and validate migrated data)
+1. re-configure masters to use the etcd3 storage backend
+1. start master daemons
+
+It is highly recommended to perform all the steps automatically.
+If any of the etcd members is started too late, the member will not get accepted by the cluster.
+
+== Deployments
+
+The etcd migration is supported over the following deployments:
+
+* rpm based deployment
+* container based deployment
+
```

Comment:
Yeah, we'll need to. We should be able to run this inside the master container. Lets sync up and see if we can get that implemented.